File: FENCE/fence_for.cpp
#include <stdio.h>

#define TRUE    (0)
#define FALSE   (1)

int solve( int * aFenceArray, 
           int   aMaxValue,
           int   aListCount )
{
    int sCount = 0;
    int sContinue = TRUE;
    int sIndex = 0;
    int sContinueCount = 0;
    int sTempMaxSize = 0;
    int sMaxSize = 0;
    int sSize = 0;

    sSize = aMaxValue;

    while ( sContinue == TRUE )
    {
        for ( sIndex = 0; sIndex < aListCount; sIndex++ )
        {
            if ( aFenceArray[sIndex] >= sSize )
            {
                sContinueCount++;
            }
            else
            {
                sTempMaxSize = sContinueCount * sSize;

                if ( sMaxSize < sTempMaxSize )
                {
                    sMaxSize = sTempMaxSize;
                }

                sContinueCount = 0;
                sTempMaxSize = 0;
            }
        }

        sTempMaxSize = sContinueCount * sSize;
        if ( sMaxSize < sTempMaxSize )
        {
            sMaxSize = sTempMaxSize;
        }

        sContinueCount = 0;
        sTempMaxSize = 0;

        sSize--;

        if ( sSize == 0 )
        {
            sContinue = FALSE;
        }

        sTempMaxSize = sSize * aListCount;
        if ( sTempMaxSize < sMaxSize )
        {
            sContinue = FALSE;
        }
    }

    return sMaxSize;
}

int main( void )
{
    int sCount = 0;
    int sListCount = 0;
    int sMaxValue = 0;
    int sFenceArray[20000];
    int i = 0;
    int j = 0;

    scanf( "%d", &sCount );

    for ( i = 0; i < sCount; i++ )
    {
        scanf( "%d", &sListCount );

        for ( j = 0; j < sListCount; j++ )
        {
            scanf( "%d ", &sFenceArray[j] );

            if ( sFenceArray[j] > sMaxValue )
            {
                sMaxValue = sFenceArray[j];
            }
        }

        printf( "%d\n", solve( sFenceArray,
                               sMaxValue,
                               sListCount ) );
                               
    }

    return 0;
}

File: FENCE/fence_for.py
`def solve( aFenceList, aListCount ):

    sTempMaxSize = 0
    sMaxSize = 0
    sContinue = True
    sContinueCount = 0

    sSize = max( aFenceList )

    #print "MaxSIze = ", sSize
    #print aFenceList
    while ( sContinue ) :

        #sContinue = False
        sTempMaxSize = 0

        for sIndex in xrange(0, len(aFenceList)) :
            #print sIndex, sSize, aFenceList[sIndex]
            #print aFenceList
            if aFenceList[sIndex] > sSize :
                sContinueCount = sContinueCount + 1
                #sContinue = True
            elif aFenceList[sIndex] == sSize :
                sContinueCount = sContinueCount + 1
                #sContinue = True
            else :
                sTempMaxSize = sContinueCount * sSize

                if sMaxSize < sTempMaxSize :
                    #print "DEBUG ", sIndex, sSize, aFenceList[sIndex]
                    #print "MAX Size = ", sTempMaxSize, sContinueCount
                    sMaxSize = sTempMaxSize;

                sContinueCount = 0

        sTempMaxSize = sContinueCount * sSize

        if sMaxSize < sTempMaxSize :
            #print "DEBUG ", sIndex, sSize, aFenceList[sIndex]
            #print "MAX Size = ", sTempMaxSize, sContinueCount
            sMaxSize = sTempMaxSize;
            sContinueCount = 0

        sSize = sSize - 1
        sContinueCount = 0

        if sSize < 1 :
            sContinue = False

        sTempMaxSize = sSize * aListCount
        if sTempMaxSize < sMaxSize :
            sContinue = False

    return sMaxSize

def main() :
    sCount = input()

    for i  in range( sCount ) :
        sNumCount = input()
        sInput = raw_input()
        sFenceList = sInput.split()

        sFenceList = [int(num) for num in sFenceList ]

        print solve( sFenceList, sNumCount )

main()
